src/test/ui/type-ascription/as_ref.stderr

```diff
@@ -0,0 +1,46 @@
+error: type ascriptions are not allowed in lvalue contexts
+  --> $DIR/as_ref.rs:19:11
+   |
+LL |   let x = (r : &Foo).as_ref();
+   |           ^^^^^^^^^^
+   |
+help: try to use the type ascription when creating the local variable
+   |
+LL | let r: &Foo = &mut foo;
+   | ^^^^^^^^^^^
+
+error: type ascriptions are not allowed in lvalue contexts
+  --> $DIR/as_ref.rs:22:21
+   |
+LL |   let another_one = (r : &Foo).as_ref().a;
+   |                     ^^^^^^^^^^
+   |
+help: try to use the type ascription when creating the local variable
+   |
+LL | let r: &Foo = &mut foo;
+   | ^^^^^^^^^^^
+
+error: type ascriptions are not allowed in lvalue contexts
+  --> $DIR/as_ref.rs:25:21
+   |
+LL |   let last_one = &*((r : &Foo).as_ref());
+   |                     ^^^^^^^^^^
+   |
+help: try to use the type ascription when creating the local variable
+   |
+LL | let r: &Foo = &mut foo;
+   | ^^^^^^^^^^^
+
+error: type ascriptions are not allowed in lvalue contexts
```

Comment:
I think we can fix these repetitions of errors caused by nested `delegate.borrow` calls using the hir visitor. I want to see whether the general logic is correct now, before doing that.